textures/src/main/java/com/dumbdogdiner/stickyapi/util/textures/TextureValidator.java

```diff
@@ -0,0 +1,172 @@
+        try {
+            // Make sure URL scheme is valid
+            if (!UrlValidator.getInstance().isValid(textureURL)) {
+                throw new MalformedURLException("The texture URL " + textureURL + " is not a valid URL");
+            }
+
+            // Needs to be from minecraft.net or mojang.com
+            @NotNull String host = HttpUrl.get(textureURL).host().toLowerCase();
+            if (!(host.endsWith("mojang.com") || host.endsWith("minecraft.net"))) {
+                throw new MalformedURLException("The texture URL " + textureURL + " specified a host other than minecraft.net or mojang.com");
+            }
+
+            // Needs to be resolvable
+            @NotNull Response resp = TextureHelper.httpClient.newCall(new Request.Builder().url(textureURL).build()).execute();
+            if (resp.code() != 200) {
+                resp.close();
+                throw new Exception("Non 200 OK Response of " + resp.code() + " received from " + textureURL);
+            }
+
+            // Needs to have a content type of PNG
+            if (!(resp.header("Content-Type").toLowerCase().startsWith("image/png") || resp.header("Content-Type").toLowerCase().startsWith("binary/octet-stream"))) {
+                resp.close();
+                throw new Exception("Unexpected Content-Type of " + resp.header("Content-Type") + " was received from " + textureURL);
+            }
+
+            // Verify that the image is, in fact, an actual PNG, and a valid PNG
+            BufferedImage img;
+            try {
+                byte[] response = resp.body().bytes();
+
+                if ((img = ImageIO.read(new ByteArrayInputStream(response))) != null) {
+                    for (int i = 0; i < PNG_SIGNATURE.length; i++) {
+                        if (PNG_SIGNATURE[i] != response[i]) {
+                            int j = i + 1;
+                            throw new Exception("Byte " + j + " of the response (" + response[i] + ") did not match byte " + j + " of the PNG signature (" + PNG_SIGNATURE[i] + ")");
+                        }
+                    }
+
+                } else {
+                    throw new NullArgumentException("The image retrieved from " + textureURL + " was decoded to null and" /* must not be null */);
+                }
+            } catch (IOException | NullArgumentException | NullPointerException e) {
+                throw new Exception("The content retrieved from " + textureURL + " was not a recognized image, was null, or was decoded to null", e);
+            }
+
+            // Needs to be 64x32 or 64x64
+            Preconditions.checkArgument(img.getWidth() == 64 && (img.getHeight() == 64 || img.getHeight() == 32),
+                    "The texture must be either 64x64 (for a new texture) or 64x32 (for a classic texture)");
+
+
+        } catch (Exception e) {
+            throw new InvalidTextureException(textureURL, e);
```

Comment:
Is a try-catch really necessary if you're just going to throw the error anyway?

Reply:
Yes for rewrapping as a runtime exception